src/library_workerfs.js

```diff
@@ -0,0 +1,95 @@
+mergeInto(LibraryManager.library, {
+  $WORKERFS__deps: ['$FS'],
+  $WORKERFS: {
+    DIR_MODE: {{{ cDefine('S_IFDIR') }}} | 511 /* 0777 */,
+    FILE_MODE: {{{ cDefine('S_IFREG') }}} | 511 /* 0777 */,
+    mount: function (mount) {
+      assert(ENVIRONMENT_IS_WORKER);
+      var root = WORKERFS.createNode(null, '/', WORKERFS.DIR_MODE, 0);
+      (mount.opts.files || []).forEach(function(file) {
```

Comment:
Also, File's name property [always contain only basename](https://developer.mozilla.org/en-US/docs/Web/API/File/name). There is no directories.